### PR TITLE
stack.c : some changes and improvements

### PR DESCRIPTION
**Pull request by christianbender**
**Changes:**

* Used the ```assert``` makro for preconditions.  
* Changed the formatting according to the [code style conventions](https://github.com/christianbender/cosmos/tree/master/guides/coding_style/c)  
* Added a function ```delete_stack(...)``` for tidy up the Heap.  
* Changed the ```main```

### CODE REVIEW

File: code/data_structures/src/stack/stack/stack.c [hunk 1]
```diff
@@ -9,45 +10,63 @@ struct Stack
...
+create_stack(unsigned capacity)
 {
     struct Stack* stack = (struct Stack*) malloc(sizeof(struct Stack));
     stack->capacity = capacity;
     stack->top = -1;
     stack->array = (int*) malloc(stack->capacity * sizeof(int));
     return stack;
 }
+
+void delete_stack(struct Stack * stack)
```
Comment (arnavb) [inline code/data_structures/src/stack/stack/stack.c]: void should go ints own line as well.

File: code/data_structures/src/stack/stack/stack.c [hunk 2]
```diff
@@ -9,45 +10,63 @@ struct Stack
...
  
-struct Stack* createStack(unsigned capacity)
+struct Stack* 
+create_stack(unsigned capacity)
 {
     struct Stack* stack = (struct Stack*) malloc(sizeof(struct Stack));
     stack->capacity = capacity;
     stack->top = -1;
     stack->array = (int*) malloc(stack->capacity * sizeof(int));
     return stack;
```
Comment (arnavb) [inline code/data_structures/src/stack/stack/stack.c]: Could you add parentheses around the return type?

File: code/data_structures/src/stack/stack/stack.c [hunk 3]
```diff
@@ -9,45 +10,63 @@ struct Stack
...
+    free(stack->array);
+    free(stack);
+}
  
-int isFull(struct Stack* stack)
-{   return stack->top == stack->capacity - 1; }
+int 
+is_full(struct Stack* stack)
+{   
+    return stack->top == stack->capacity - 1;
```
Comment (arnavb) [inline code/data_structures/src/stack/stack/stack.c]: returns go in parentheses.

**Comment by christianbender** [comment]
@arnavb Done. I have changed the formatting as requested.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**
positive_opengenus_cosmos

**Pull request merged**